staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -2584,3 +2598,22 @@ func watchAndWaitForBookmark(t *testing.T, ctx context.Context, etcdStorage stor
 		return rv
 	}
 }
+
+// TODO(p0lyn0mial): fix me, temporalHackForTestsThatRequireStorageRequestWatchProgress
+// inits the storage layer so that tests that require storage.RequestWatchProgress pass
+//
+// In the future we could have a function that would allow for setting the feature
+// only for duration of a test.
+func temporalHackForTestsThatRequireStorageRequestWatchProgress(t *testing.T) {
```

Comment:
Let's avoid naming like that... maybe call it `forceRequestWatchProgressSupport`

Also - instead of creating etcd here, how about
```
oldFeatureSupportChecker = DefaultFeatureSupportChecker
defer func() { DefaultFeatureSupportChecker = oldFeatureSupportChecker }
DefaultFeatureSupportChecker = &supportsEverythingSupportChecker{}
```

Still not beautiful, but I think it's better...

Reply:
I would like to avoid changing `DefaultFeatureSupportChecker` without any coordination, as it may cause issues for other tests. Additionally, I would like to avoid replacing the global variable altogether. I think we could implement a mechanism similar to the one we have for feature gates, allowing a test for a feature mutation for the duration of the test.